salt/modules/x509.py

```diff
@@ -41,7 +41,7 @@
 from salt.utils.odict import OrderedDict
 
 try:
-    import M2Crypto
+    from M2Crypto import ASN1, BIO, EVP, RSA, X509, m2
```

Comment:
Nitpick: not sure if I'm missing something here, but if we do like this:

```suggestion
    import M2Crypto
    from M2Crypto import ASN1, BIO, EVP, RSA, X509, m2
```

This would also keep the compatiblity between 0.44.0 and lower versions, and will prevent the rest changes on this file.

Reply:
How could it help? `M2Crypto` with this change is not used anymore in the code.

Reply:
I agree this is just cosmetic.

With the suggested changes, we don't need to change the rest of the file and we can just keep the calls as they are right now, i.a. `M2Crypto.X509.X509_Extension` (keeping the `M2Crypto` prefix), instead of `X509.X509_Extension`.

Reply:
It will not work that way. In case of using `import M2Crypto` it can use `M2Crypto.X509.X509_Extension` only if they are defined inside `M2Crypto/__init__.py`, but with the recent version of `M2Crypto` they are not defined there, but `from M2Crypto import ASN1, BIO, EVP, RSA, X509, m2` is not allowing to use `M2Crypto.X509.X509_Extension` as it loads `X509` as standalone definition which should be used as `X509.X509_Extension`, but not as `M2Crypto.X509...`

Reply:
With this change both will work, but the key point here that `import M2Crypto` became useless as not used at all in the code. And BTD, if you've tested it with the bundle in the `next` there is a roll back patch for it applied for `M2Crypto` that's why it's not failing with old way of using `M2Crypto`. Because of this: https://build.opensuse.org/projects/home:vizhestkov:new-salt-deps/packages/saltbundlepy-m2crypto/files/fix-imcompatibility-with-salt.patch?expand=1